### DISCUSSION TITLE
How to tell homebrew I've installed a dependency manually?

### DISCUSSION THREAD

**Discussion by TC72** [Homebrew/discussions#331] [Everyday usage]
This is related to M1 chip just because of the current state of formulas while the transfer is happening, but the question isn't specific to M1.

I've tried to install ripgrep which relies on rust and it just never finishes. I decided to use rustup to install rust instead. It worked fine when I told it to use the nightly and I now have rust installed in ~/.cargo/bin.
```
rustc 1.50.0-nightly (1f5bc176b 2020-12-19)
```
So tried to use brew to install ripgrep again and it's trying to install rust again because it doesn't know I've installed rust manually.

Is there anyway  to tell brew I've done this and it doesn't need to install rust?

**Comment by carlocab** [comment]
A quick-and-dirty solution might be to use `brew install --ignore-dependencies`. You may also need to pass `--env=std` so that `rustc` can be found in your build environment. The downside to this solution is that it might break on you in subtle and unexpected ways. 

A better solution is to maintain your own tap for the formulae whose dependencies you've switched out. (And possibly also for any of your own custom dependencies, like the nightly version of rust.)

Another might be to try `brew install --HEAD rust`. `brew` won't automatically upgrade formulae that were installed with `--HEAD`, so anything you've installed that depends on the nightly version of rust (probably) won't break (unless it's a bug in rust). However, the formula was not written with the intention of `--HEAD` working, so this may not work.

**Reply by jimtut** [reply]
I was about to post a similar question, also about ignoring a locally installed rust.

I tried `brew install -s --ignore-dependencies --env=std ncspot` and got
```
Warning: --ignore-dependencies is an unsupported Homebrew developer flag!
Adjust your PATH to put any preferred versions of applications earlier in the
PATH rather than using this unsupported flag!
...
Error: An exception occurred within a child process:
  RuntimeError: /opt/homebrew/opt/rust not present or broken
Please reinstall rust. Sorry :(
```
Tried again with `brew install --HEAD -s rust` and got:
```
==> ./configure --prefix=/opt/homebrew/Cellar/rust/HEAD-2ad5292 --disable-rpath --release-channel=nightly --build=x86_64
==> arch -x86_64 make
Last 15 lines from /Users/jim/Library/Logs/Homebrew/rust/02.arch:
               -exported_symbol[s_list] command line option
          ld: symbol(s) not found for architecture x86_64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
          

error: aborting due to previous error

error: could not compile `std`

To learn more, run the command again with --verbose.
command did not execute successfully: "/private/tmp/rust-20201220-66839-166nlms/build/x86_64-apple-darwin/stage0/bin/cargo" "build" "--target" "aarch64-apple-darwin" "-Zbinary-dep-depinfo" "-j" "8" "--release" "--features" "panic-unwind backtrace compiler-builtins-c" "--manifest-path" "/private/tmp/rust-20201220-66839-166nlms/library/test/Cargo.toml" "--message-format" "json-render-diagnostics"
expected success, got: exit code: 101
failed to run: /private/tmp/rust-20201220-66839-166nlms/build/bootstrap/debug/bootstrap build --stage 2
Build completed unsuccessfully in 0:34:31
make: *** [all] Error 1
```

Rust via homebrew on M1 seems very broken, but it seems to be working out-of-the-box (without using homebrew to install it), although I'm still unable to install this ncspot, using homebrew or cargo...

**Comment by TC72** [comment]
Thanks Carlo, I'll give these a try.

**Comment by J-M0** [comment]
Since you're already compiling ripgrep from source and you've installed rust manually, why not install it via cargo?
